public/app/features/explore/utils/links.ts

```diff
@@ -72,22 +72,37 @@ export const getFieldLinksForExplore = (options: {
     text: 'Raw value',
   };
 
+  let fieldDisplayValuesProxy;
+
   // If we have a dataFrame we can allow referencing other columns and their values in the interpolation.
   if (dataFrame) {
+    fieldDisplayValuesProxy = getFieldDisplayValuesProxy({
+      frame: dataFrame,
+      rowIndex,
+    });
+
     scopedVars['__data'] = {
       value: {
         name: dataFrame.name,
         refId: dataFrame.refId,
-        fields: getFieldDisplayValuesProxy({
-          frame: dataFrame,
-          rowIndex,
-        }),
+        fields: fieldDisplayValuesProxy,
       },
       text: 'Data',
     };
   }
 
   if (field.config.links) {
+    if (fieldDisplayValuesProxy && fieldDisplayValuesProxy[field.name] !== undefined) {
+      scopedVars['__targetField'] = {
+        value: fieldDisplayValuesProxy[field.name],
+        text: fieldDisplayValuesProxy[field.name],
```

Comment:
I think `text` is not needed here if it's the same as `value`. Interpolation logic uses it to override `value` if needed -> https://github.com/grafana/grafana/blob/main/public/app/features/templating/template_srv.ts#L169